File: RecyclerView/app/src/main/java/com/xamfy/recyclerview/MainActivity.java
package com.xamfy.recyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.facebook.shimmer.ShimmerFrameLayout;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    RecyclerView recyclerView;
    private ShimmerFrameLayout mShimmerFrameLayout;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(TAG, "onCreate: started");
        mShimmerFrameLayout = findViewById(R.id.shimmer_view_container);
        recyclerView = findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        getData();
    }

    @Override
    public void onResume() {
        super.onResume();
        mShimmerFrameLayout.startShimmer();
    }

    @Override
    protected void onPause() {
        mShimmerFrameLayout.stopShimmer();
        super.onPause();
    }

    private void getData() {
        Call<List<ImageList>> imageList = API.getService().getImageList();
        imageList.enqueue(new Callback<List<ImageList>>() {
            @Override
            public void onResponse(Call<List<ImageList>> call, Response<List<ImageList>> response) {
                List<ImageList> list = response.body();
                recyclerView.setAdapter(new RecyclerViewAdapter(list, MainActivity.this));
                mShimmerFrameLayout.stopShimmer();
                mShimmerFrameLayout.setVisibility(View.GONE);
//                Toast.makeText(MainActivity.this,   "Success", Toast.LENGTH_SHORT).show();
//                Log.d(TAG, "onResponse: "+list.get(0).getName());
            }

            @Override
            public void onFailure(Call<List<ImageList>> call, Throwable t) {
                Toast.makeText(MainActivity.this, "Error occurred", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: JetPack/Architecture-Components/All/DataBinding/app/src/main/java/com/xamfy/databinding/MainActivity.kt
package com.xamfy.databinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.xamfy.databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val binding : ActivityMainBinding = DataBindingUtil.setContentView(this,
            R.layout.activity_main)

//        binding.tvName.text = "xamfy"
//        binding.tvEmail.text = "xamfy@example.com"
        binding.contact = Contact("xamfy", "xamfy@example.com")
        binding.handler = EventHandler(this)
        binding.imageUrl = "https://i.redd.it/lhw4vp5yoy121.jpg"

    }
}
